src/kalman/kalmanObject.cpp

```diff
@@ -166,5 +214,18 @@ namespace rtt {
         }
     }
 
+    float kalmanObject::calculateRot(float obsRot){
+        float rotDiff = this->X(4)-obsRot;
```

Comment:
This function will give problems with floating point errors if your angle is exactly -pi. You need to a the numeric limit of float to the lowest value in that case.

Reply:
what you mean angle exactly -pi? the difference?